packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -763,7 +789,7 @@ export class RestApi extends RestApiBase {
       description: props.description,
       policy: props.policy,
       failOnWarnings: props.failOnWarnings,
-      minimumCompressionSize: props.minimumCompressionSize,
+      minimumCompressionSize: props.minCompressionSize?.toBytes() || props.minimumCompressionSize,
```

Comment:
For *reasons* (gestures wildly), this will not what you expect it to do when you pass in `Size.bytes(0)`. 

* Try it first in a new unit test
* Then read about it
* Then replace it with:
```suggestion
      minimumCompressionSize: props.minCompressionSize?.toBytes() ?? props.minimumCompressionSize,
```